command/agent/command.go

```diff
@@ -134,10 +165,26 @@ func (c *Command) readConfig() *Config {
 		return config
 	}
 
-	// Check that we have a data-dir
-	if config.DataDir == "" {
+	// Check for valid modes.
+	if config.Server.Enabled && config.Client.Enabled {
+		c.Ui.Error("To run both as a server and client, use -dev mode.")
```

Comment:
I guess that is true. If you wanted to "dev" mode with persistence you could enable both. I will update.